lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -632,25 +634,36 @@ def install_puppet_from_dmg_on( hosts, opts )
         end
         alias_method :install_puppet_from_dmg, :install_puppet_from_dmg_on
 
-        # Installs Puppet and dependencies from dmg on provided host(s).
+        # Installs puppet-agent and dependencies from dmg on provided host(s).
         #
         # @param [Host, Array<Host>, String, Symbol] hosts    One or more hosts to act upon,
         #                            or a role (String or Symbol) that identifies one or more hosts.
         # @param [Hash{Symbol=>String}] opts An options hash
-        # @option opts [String] :puppet_agent_version The version of Puppet Agent to install
-        # @option opts [String] :mac_download_url Url to download msi pattern of %url%/puppet-%version%.msi
+        # @option opts [String] :puppet_agent_version The version of Puppet Agent to install, defaults to latest
+        # @option opts [String] :mac_download_url Url to download msi pattern of %url%/puppet-%version%.dmg
+        # @option opts [String] :puppet_collection Defaults to 'PC1'
         #
         # @return nil
         # @api private
         def install_puppet_agent_from_dmg_on(hosts, opts)
+          opts[:puppet_collection] ||= 'PC1'
+          opts[:puppet_collection] = opts[:puppet_collection].upcase #needs to be upcase, more lovely consistency
           block_on hosts do |host|
 
             host[:type] = 'aio' #we are installing agent, so we want aio type
 
-            version = opts[:puppet_agent_version] || 'latest'
-            on host, "curl -O #{opts[:mac_download_url]}/puppet-agent-#{version}.dmg"
+            variant, version, arch, codename = host['platform'].to_array
+            agent_version = opts[:puppet_agent_version] || 'latest'
+            pkg_name = 'puppet-agent-*'
+            if agent_version == 'latest'
+              dmg_name = "puppet-agent-#{agent_version}.dmg"
+              on host, "curl -O #{opts[:mac_download_url]}/#{dmg_name}"
+            else
+              dmg_name = "puppet-agent-#{agent_version}-osx-#{version}-x86_64.dmg"
+              on host, "curl -O #{opts[:mac_download_url]}/#{opts[:puppet_collection]}/#{dmg_name}"
+            end
 
-            host.install_package("puppet-agent-#{version}")
+            host.install_package(pkg_name)
```

Comment:
dmg_name actually won't work because the installer name and the package name that it installs are different - that was the first approach that I took.